lib/checkcondition.cpp

```diff
@@ -1490,6 +1490,8 @@ void CheckCondition::alwaysTrueFalse()
                 else if (parent->str() == ";" && parent->astParent() && parent->astParent()->astParent() &&
                          Token::simpleMatch(parent->astParent()->astParent()->previous(), "for ("))
                     condition = parent->astParent()->astParent()->previous();
+                else if (isBooleanFuncArg(tok))
```

Comment:
This should probably go into the `knownArgument` checker.

Reply:
Looking at the test cases, `knownArgument` seems to be more about calculations with a known result, not simple conditions.

Reply:
It checks comparisons as well, but only for integral variables.